servers/src/main/java/tachyon/master/file/FileSystemMaster.java

```diff
@@ -1392,6 +1395,65 @@ public void resetFile(long fileId) throws FileDoesNotExistException {
   }
 
   /**
+   * Sets TTL of a file no matter whether the file already has a TTL.
+   *
+   * <p>
+   * If the file already has a TTL, and the new TTL doesn't equal {@link Constants#NO_TTL}, the
+   * corresponding {@link InodeFile} in {@link #mTTLBuckets} will be moved to a new appropriate
+   * bucket according to the new TTL.
+   * <p>
+   * If the file already has a TTL, and the new TTL equals {@link Constants#NO_TTL}, the
+   * corresponding {@link InodeFile} will be moved out of {@link #mTTLBuckets}.
+   * <p>
+   * If the file doesn't have a TTL before, and the new TTL doesn't equal {@link Constants#NO_TTL},
+   * then the new TTL will be set and the corresponding {@link InodeFile} will be added to
+   * {@link #mTTLBuckets}.
+   *
+   * @param fileId the id of the file
+   * @param ttl the new ttl value to be set for the file
+   * @throws FileDoesNotExistException if the file doesn't exist
+   */
+  public void setTTL(long fileId, long ttl) throws FileDoesNotExistException {
+    synchronized (mInodeTree) {
+      setTTLInternal(fileId, ttl);
+      writeJournalEntry(new SetTTLEntry(fileId, ttl));
+    }
+  }
+
+  private void setTTLFromEntry(SetTTLEntry entry) {
+    try {
+      setTTLInternal(entry.getId(), entry.getTTL());
+    } catch (FileDoesNotExistException fdnee) {
```

Comment:
for consistency with other `*FromEntry` methods, catch this exception in `processJournalEntry`